lib/ansible/galaxy/collection.py

```diff
@@ -382,10 +382,16 @@ def publish_collection(collection_path, api, wait, timeout):
     :param timeout: The time in seconds to wait for the import process to finish, 0 is indefinite.
     """
     import_uri = api.publish_collection(collection_path)
+    # Galaxy returns a url fragment which differs between v2 and v3.  The second to last entry is
+    # always the task_id, though.
+    # v2: {"task": "https://galaxy-dev.ansible.com/api/v2/collection-imports/35573/"}
+    # v3: {"task": "/api/automation-hub/v3/imports/collections/838d1308-a8f4-402c-95cb-7823f3806cd8/"}
```

Comment:
Are we 100% positive we always get a trailing `/` on `import_uri`? If the trailing `/` is missing, we'll get the incorrect value for `task_id`.

Reply:
It would be a change in the API but that bothered me too.  I can code something more robust.